src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+
+/**
+ * Parses in the user input and executes the corresponding actions
+ */
+public class Parser {
+    private final String command;
+    private final Ui ui;
+
+    /**
+     * Constructor for parser
+     * @param command user's inputted command
+     */
+    public Parser(String command) {
+        this.command = command;
+        this.ui = new Ui();
+    }
+
+    /**
+     * Parses the command and matches it with the corresponding action to be taken
+     * Executes the action on the task list given
+     * @param tasks ArrayList of current tasks stored on the chatbot
+     * @return boolean true if bye command entered and chatbot should be stopped, false otherwise
+     */
+    public String executeCommand(TaskList tasks, Storage storage) throws IOException {
+        String[] commands = command.strip().split(" ", 2);
+        try {
+            switch (commands[0]) {
+            case "bye":
+                storage.saveTasks(tasks);
+                return ui.showBye();
+            case "list":
+                return ui.getCorrectMessage(Ui.Commands.LIST, tasks, 0);
+            case "mark":
+                if (commands.length == 1) {
+                    throw new EmptyDescriptionException("mark");
```

Comment:
Perhaps you can write a method for this so you don't have to repeat it for other commands?